Add AMPS support entry (#14250)
symfony/symfony-docs: messenger.rst

AMQP Transport
~~~~~~~~~~~~~~

.. versionadded:: 5.2

>     Starting from Symfony 5.2, the AMQP transport can handle AMQPS DSN (e.g.  ``amqps://``).

Reviewer: ```suggestion
    Starting from Symfony 5.2, the AMQP transport can handle AMQPS DSN (e.g. ``amqps://``).
```

>     Be aware that using it without using CA certificate can throw an exception.

Reviewer: ```suggestion
    Be aware that using it without using a CA certificate can throw an exception.
```

>     Be aware that using it without using CA certificate can throw an exception.

Reviewer 2: This sentence isn't really helpful. How do I set this CA certificate? We should either not mention it (and rely on Symfony to throw an exception that will educate the user on what to do next) or fully show how to use amps.

Author: I've added an entry about it, wdyt ?

>     An alternative is to use AMQP DSN and to specify which port you want to use

Reviewer 3: ```suggestion
    An alternative is to use AMQP DSN and specify the port to use.
```

>     An alternative is to use AMQP DSN and to specify which port you want to use (e.g.  ``amqp://localhost?port=5671``).

Reviewer 4: We often avoid introducing new information in versionadded directives. When the version reached end-of-life, we simply remove all old versionadded directives - it would be sad to loose usefull information.

So I would suggest you say this information somewhere below in the text. I think the best is to add a real example to the `.env` example on line 866, e.g.:

```
# .env
MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages

# or use the AMQPS protocol
MESSENGER_TRANSPORT_DSN=amqps://...
```

You can then also show a working DNS string (so people see how to configure the required CA certificate). After the example, you can then add a short versionadded directive:

```rst
.. versionadded:: 5.2

    Support for the AMQPS DSN was introduced in Symfony 5.2.
```

Author: Wdyt about the changes ?

    # .env
    MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages

    # or use the AMQPS protocol
    MESSENGER_TRANSPORT_DSN=amqps://guest:guest@localhost/%2f/messages

To use Symfony's built-in AMQP transport, you need the AMQP PHP extension.

> If you want to use TLS/SSL encrypted AMQP you must provide a CA certificate, you should set it through ``amqp.cacert = /etc/ssl/certs`` (path depend of your system) to your ``php.ini`` file or by setting the ``cacert`` setting (e.g ``amqps://localhost?cacert=/etc/ssl/certs/``)

Reviewer 5: I'm not a native speaker so please ignore if im wrong but it think it should 
```suggestion
If you want to use TLS/SSL encrypted AMQP you must provide a CA certificate. You should set it through ``amqp.cacert = /etc/ssl/certs`` (path depends on your system) in your ``php.ini`` file or by setting the ``cacert`` setting (e.g ``amqps://localhost?cacert=/etc/ssl/certs/``)
```

> If you want to use TLS/SSL encrypted AMQP you must provide a CA certificate. You should set it through ``amqp.cacert = /etc/ssl/certs`` (path depends on your system) in your ``php.ini`` file or by setting the ``cacert`` setting (e.g ``amqps://localhost?cacert=/etc/ssl/certs/``)

Reviewer 6: ```suggestion
If you want to use TLS/SSL encrypted AMQP you must provide a CA certificate. You need to set it using ``amqp.cacert = /etc/ssl/certs`` (path depends on your system) in your ``php.ini`` file or by setting the ``cacert`` parameter (e.g ``amqps://localhost?cacert=/etc/ssl/certs/``)
```

Author: Setting is actually used in the same file below. In think we should use the same word for the same thing no ?

Reviewer 6: When dealing with URIs with query strings, the word "setting" is never used for anything after the `?`, and I find it confusing in a Symfony context, as a setting is something to go in a config file.

RFC3986 refers to
> "key=value" pairs
while [Wikipedia](https://en.wikipedia.org/wiki/Query_string) talks of
> multiple parameters in the query string, separated by some delimiter

In the end it comes down to terminology that Symfony chooses to use. My original reason for the edit was having two 'setting's in one sentence isn't considered good English :)

> Using TLS/SSL encrypted AMQP DSN will use port 5671. You can overwrite this behavior by setting the ``port`` setting (e.g. ``amqps://localhost?cacert=/etc/ssl/certs/&port=12345``).

Reviewer 6: I assume this isn't a Symfony only setting, and AMQP always uses port 5671 by default?
```suggestion
By default TLS/SSL encrypted AMQP uses port 5671. You can overwrite this behavior by setting the ``port`` parameter (e.g. ``amqps://localhost?cacert=/etc/ssl/certs/&port=12345``).
```

> By default TLS/SSL encrypted AMQP uses port 5671. You can overwrite this behavior by setting the ``port`` parameter (e.g. ``amqps://localhost?cacert=/etc/ssl/certs/&port=12345``).

Reviewer 7: is American or British english a convention used across the docs?
🇺🇸  `behavior` vs  🇬🇧  `behaviour`

Author: javier told me about using behavior

Reviewer 8: Yes, we use American English (see https://symfony.com/doc/current/contributing/documentation/standards.html)
